lib/medium-client.js

```diff
@@ -0,0 +1,423 @@
+ *
+ * @param {string} code
+ * @param {string} redirectUrl
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.exchangeAuthorizationCode = function (code, redirectUrl, callback) {
+  this._acquireAccessToken({
+    code: code,
+    client_id: this._clientId,
+    client_secret: this._clientSecret,
+    grant_type: 'authorization_code',
+    redirect_uri: redirectUrl
+  }, callback)
+}
+
+
+/**
+ * Exchanges a refresh token for an access token and a refresh token.
+ *
+ * @param {string} refreshToken
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.exchangeRefreshToken = function (refreshToken, callback) {
+  this._acquireAccessToken({
+    refresh_token: refreshToken,
+    client_id: this._clientId,
+    client_secret: this._clientSecret,
+    grant_type: 'refresh_token'
+  }, callback)
+}
+
+
+/**
+ * Returns the details of the user associated with the current
+ * access token.
+ *
+ * Requires the current access token to have the basicProfile scope.
+ *
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.getUser = function (callback) {
+  this._makeRequest({
+    method: 'GET',
+    path: '/v1/me'
+  }, callback)
+}
+
+
+/**
+ * Returns the publications related to the current user. Notice that
+ * the userId needs to be passed in as an option. It can be acquired
+ * with a call to getUser().
+ *
+ * Requires the current access token to have the
+ * listPublications scope.
+ *
+ * @param {{
+ *  userId: string
+ * }}
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.getPublicationsForUser = function (options, callback) {
+  this._enforce(options, ['userId'])
+  this._makeRequest({
+    method: 'GET',
+    path: '/v1/users/' + options.userId + '/publications'
+  }, callback)
+}
+
+
+/**
+ * Returns the contributors for a chosen publication. The publication is identified
+ * by the publication ID included in the options argument. IDs for publications
+ * can be acquired by getUsersPublications.
+ *
+ * Requires the current access token to have the basicProfile scope.
+ *
+ * @param {{
+ *  publicationId: string
+ * }}
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.getContributorsForPublication = function (options, callback) {
+  this._enforce(options, ['publicationId'])
+  this._makeRequest({
+    method: 'GET',
+    path: '/v1/publications/' + options.publicationId + '/contributors'
+  }, callback)
+}
+
+
+/**
+ * Creates a post on Medium.
+ *
+ * Requires the current access token to have the publishPost scope.
+ *
+ * @param {{
+ *  userId: string,
+ *  title: string,
+ *  contentFormat: PostContentFormat,
+ *  content: string,
+ *  tags: Array.<string>,
+ *  canonicalUrl: string,
+ *  publishStatus: PostPublishStatus,
+ *  license: PostLicense
+ * }} options
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.createPost = function (options, callback) {
+  this._enforce(options, ['title', 'contentFormat', 'content'])
```

Comment:
Changed the use of _enforce so that the code only calls it on parameters that the *client* code requires, leaving enforcement of parameters that the server needs for the server